polygon/bor/bor.go

```diff
@@ -1317,7 +1317,7 @@ func (c *Bor) checkAndCommitSpan(
 
 	// check span is not set initially
 	if currentSpan.EndBlock == 0 {
-		return c.fetchAndCommitSpan(currentSpan.ID, state, header, chain, syscall)
+		return c.fetchAndCommitSpan(currentSpan.ID+1, state, header, chain, syscall)
```

Comment:
please leave a comment explaining why it is +1 here

Reply:
Sure. A simpler intuitive understanding is it's equivalence with what's present in bor as seen [here](https://github.com/maticnetwork/bor/blob/v1.2.3/consensus/bor/bor.go#L1150-L1152). We always fetch the next span based on current span's ID.

A more detailed understanding is as follows: So this function `checkAndCommitSpan` is called at start of every sprint. Simply put, whenever required, it fetches the span required for bor to proceed. It makes sure it has the next span before that span even has started. When bor is processing first block of last sprint of current span, it fetches the next span. This is ideal for all the spans except 0 and 1. Span 0 is created inside the genesis contracts present on bor so one doesn't need to explicitly commit it. On calling `commitSpan` on span 1, it eventually also commits span 0 (see [this](https://github.com/maticnetwork/genesis-contracts/blob/master/contracts/BorValidatorSet.template#L229) for more info). So when the code flow enters this function for the very first time and calls `getCurrentSpan`, it returns an empty span (with all id, start block and end block set to 0). This is what this condition specifically handles. We can't really commit span 0 as the contract won't allow us to do so. Basically, we should be committing the next span i.e. span with ID `currentSpan.ID+1` which is what this change denotes.

Reply:
@manav2401 thank you for the detailed explanation, I meant that it would be nice to leave comments in the code to guide readers why we need to have such logic